### PR TITLE
Enhancement/skale 4634 ima interfaces

### PR DESCRIPTION
**Pull request by DimaStebaev**
Move signatures of all external functions to a separate package.

### CODE REVIEW

File: proxy/contracts/test/SkaleVerifierMock.sol [hunk 1]
```diff
@@ -27,7 +27,22 @@ import "../schain/bls/FieldOperations.sol";
 import "./PrecompiledMock.sol";
 
 
-contract SkaleVerifierMock {
+interface ISkaleVerifierMock {
```
Comment (payvint) [inline proxy/contracts/test/SkaleVerifierMock.sol:30]: I guess it should be from skale-manager-interfaces

File: proxy/contracts/test/MessageProxyForSchainTester.sol [hunk 2]
```diff
@@ -23,7 +23,26 @@ pragma solidity 0.8.6;
 
 import "../schain/MessageProxyForSchain.sol";
 
-contract MessageProxyForSchainTester is MessageProxyForSchain {    
+
+interface IMessageProxyForSchainTester {
+    function postMessage(
```
Comment (payvint) [inline proxy/contracts/test/MessageProxyForSchainTester.sol:28]: Looks like it is better to take from MessageProxy interface

File: proxy/contracts/test/MessageProxyForMainnetTester.sol [hunk 3]
```diff
@@ -23,7 +23,18 @@ pragma solidity 0.8.6;
 
 import "../mainnet/MessageProxyForMainnet.sol";
 
-contract MessageProxyForMainnetTester is MessageProxyForMainnet {    
+
+interface IMessageProxyForMainnetTester {
+    function refundGasByUser(
```
Comment (payvint) [inline proxy/contracts/test/MessageProxyForMainnetTester.sol:28]: This function duplicates the same function from ICommunityPool

File: proxy/contracts/schain/tokens/ERC721OnChain.sol [hunk 4]
```diff
@@ -25,13 +25,19 @@ pragma solidity 0.8.6;
 import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721BurnableUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/token/ERC721/extensions/ERC721URIStorageUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/access/AccessControlEnumerableUpgradeable.sol";
+import "@skalenetwork/ima-interfaces/schain/tokens/IERC721OnChain.sol";
```
Comment (payvint) [inline proxy/contracts/schain/tokens/ERC721OnChain.sol:28]: There is no interfaces for such ERC721?

File: proxy/contracts/schain/tokens/ERC20OnChain.sol [hunk 5]
```diff
@@ -24,13 +24,14 @@ pragma solidity 0.8.6;
 
 import "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC20BurnableUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/access/AccessControlEnumerableUpgradeable.sol";
+import "@skalenetwork/ima-interfaces/schain/tokens/IERC20OnChain.sol";
```
Comment (payvint) [inline proxy/contracts/schain/tokens/ERC20OnChain.sol:27]: There is no interfaces for such ERC20?

File: proxy/contracts/schain/tokens/ERC1155OnChain.sol [hunk 6]
```diff
@@ -24,13 +24,14 @@ pragma solidity 0.8.6;
 
 import "@openzeppelin/contracts-upgradeable/token/ERC1155/extensions/ERC1155BurnableUpgradeable.sol";
 import "@openzeppelin/contracts-upgradeable/access/AccessControlEnumerableUpgradeable.sol";
+import "@skalenetwork/ima-interfaces/schain/tokens/IERC1155OnChain.sol";
```
Comment (payvint) [inline proxy/contracts/schain/tokens/ERC1155OnChain.sol:27]: There is no interfaces for such ERC1155?

File: proxy/contracts/schain/TokenManagerLinker.sol [hunk 7]
```diff
@@ -90,7 +95,21 @@ contract TokenManagerLinker is AccessControlEnumerableUpgradeable, IMessageRecei
      * 
      * - Function caller has to be granted with {REGISTRAR_ROLE}.
      */
-    function registerTokenManager(TokenManager newTokenManager) external onlyRegistrar {
+    function initialize(MessageProxy newMessageProxyAddress, address linker)
+        external
```
Comment (payvint) [inline proxy/contracts/schain/TokenManagerLinker.sol:94]: Why it is an external?

**Review by payvint — COMMENTED** [review]
Added questions

File: proxy/contracts/schain/TokenManagerLinker.sol (hunk 7 above)
Comment (DimaStebaev) [reply proxy/contracts/schain/TokenManagerLinker.sol:94]: It is not using from insides of the smart contract.

File: proxy/contracts/test/SkaleVerifierMock.sol (hunk 1 above)
Comment (DimaStebaev) [reply proxy/contracts/test/SkaleVerifierMock.sol:30]: Hm... It's a good idea!

Comment (DimaStebaev) [reply proxy/contracts/test/SkaleVerifierMock.sol:30]: But it's blocked by SKALE-4466.

**Review by payvint — APPROVED** [review]

**Pull request merged**